### PR TITLE
Support passing in MultiLabelBinarizer and Raw Training Data

### PR DESCRIPTION
**Pull request by adamhadani**
This PR is a clone of PR #20 which I'm using to do some cleanup on that original PR which originated with a fork there.

The primary enhancements provided by this work are:

* Support for multi-label classification via incorporating of a MultiLabelBinarizer into the classifier, along with accompanying unit-test fixtures and test cases
* Support for passing in the "raw" training data (e.g. text) rather than a pre-computed feature matrix, so that each local classifier can run its own classifier pipeline (e.g. using scikit-learn Pipeline) which opens the door to more fine-grained feature engineering and feature selection per node in the hierarchy.
* Other misc. cleanup and improvements to go along with these

This will be released as `2.0.0` version to PyPI once unit-test coverage and examples are in and this is merged.

### CODE REVIEW

File: sklearn_hierarchical_classification/tests/fixtures.py [hunk 1]
```diff
@@ -96,6 +141,50 @@ def make_classifier_and_data(
...
+        ngram_range=(1, 3),
+        max_df=1.0,
+        min_df=0.0,
+        binary=False,
+        use_idf=True,
+        smooth_idf=True,
+        sublinear_tf=True,
+        max_features=70000
+    )
+    bclf = OneVsRestClassifier(LinearSVC())
```
Comment (adamhadani) [inline sklearn_hierarchical_classification/tests/fixtures.py]: ping @fbenites  - I am very close to finalizing this PR and merging it including all your awesome work.

One pending item for me to understand - it seems like your work implicitly puts the classifier into a "LCN" (Local Classifier per Node) mode. e.g. instead of having a multi-class classifier per parent node ("LCPN"), we have a single binary classifier per node which tells you if this node should be considered as a predicated label or not.

Is this right? if so I will make some further updates / unit-tests to make sure the mlb/preprocessing flags work in both LCPN and LCN modes, as well as make the support for LCN more explicit

Comment (fbenites) [reply sklearn_hierarchical_classification/tests/fixtures.py]: It should be independent, the preprocessing can be made either way. We could create also a test with "LCPN".
